src/core/WebAppFactoryManager.cpp

```diff
@@ -72,9 +78,15 @@ WebAppFactoryInterface* WebAppFactoryManager::loadPluggable(QString appType)
         return 0;
 
     WebAppFactoryInterface* interface;
+
+#if defined(HAS_AGL_SERVICE)
+    interface = new WebAppFactoryLuna();
+    m_interfaces.insert(kDefaultAppType, interface);
+    if (!appType.isEmpty())
+        return interface;
+#else // Non-AGL
```

Comment:
I'd prefer to have WebAppFactoryManager.cpp containing only the common code. Then we'd have WebAppFactoryManagerAGL and WebAppFactoryManagerQtPlugin for the specific overrides.

We'd apply the template pattern. I.e. in this method, we'd delegate to an abstract method loadInterface() that would be implemented in each side accordingly.

About the include changes on top, the goal would be isolating the QtPlugin includes in the QtPlugin implementation. About WebAppFactoryLuna, it would be replaced by WebAppFactoryAGL and only referred in the AGL factory manager.

Reply:
Ok, agreed.

Question: Where should these new files live? `src/plugin` even they don't implement a plugin system?

Thanks.

Reply:
I think we can have the agl specific objects in src/agl. For WebAppFactoryManagerQtPlugin, it could be moved to src/plugin.